packages/graphql/tests/integration/issues/1756.int.test.ts

```diff
@@ -0,0 +1,207 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { graphql, GraphQLSchema } from "graphql";
+import type { Driver } from "neo4j-driver";
+import { Neo4jGraphQL } from "../../../src";
+import { generateUniqueType } from "../../utils/graphql-types";
+import Neo4j from "../neo4j";
+
+describe("https://github.com/neo4j/graphql/issues/1756", () => {
+    const productType = generateUniqueType("Product");
+    const genreType = generateUniqueType("Genre");
+
+    let schema: GraphQLSchema;
+    let driver: Driver;
+    let neo4j: Neo4j;
+
+    beforeAll(async () => {
+        neo4j = new Neo4j();
+        driver = await neo4j.getDriver();
+        // user bug
+        const typeDefs = `
+        interface INode {
+            id: ID! @callback(operations: [CREATE], name: "nanoid")
+        }
+        
+        type ${productType.name} implements INode {
+            id: ID!
+            name: String!
+            genre: [${genreType.name}!]! @relationship(type: "HAS_GENRE", direction: OUT)
+        }
+        
+        type ${genreType.name} implements INode {
+            id: ID!
+            value: String! @unique
+        }
+        `;
+
+        const nanoid = (_parent, _args, context) => {
+            return `callback_value`;
+        };
+        
+        const neoGraphql = new Neo4jGraphQL({ typeDefs, driver, config: { callbacks: { nanoid } } });
+        schema = await neoGraphql.getSchema();
+    });
+
+    afterAll(async () => {
+        await driver.close();
+    });
+
+    test("should not raise a GraphQL validation error if invoked without passing the id field", async () => {
+        const query = `
+        mutation {
+            ${productType.operations.create}(input: {
+              name: "TestProduct",
+              genre: {
+                connectOrCreate: [
+                  {
+                    where: {
+                      node: {
+                        value: "Action"
+                      }
+                    },
+                    onCreate: {
+                      node: {
+                        value: "Action"
+                      }
+                    }
+                  }
+                ]
+              }
+            }) {
+              ${productType.plural} {
+                id
+              }
+            }
+          }
+      `;
+
+        const result = await graphql({
+            schema,
+            source: query,
+            variableValues: {},
+            contextValue: neo4j.getContextValues(),
+        });
+
+        expect(result.errors).toBeFalsy();
+        expect(result?.data?.[productType.operations.create]).toEqual({
+            [productType.plural]: [{
+                id: "callback_value"
+            }]
+        });
+    });
+});
+
+describe("https://github.com/neo4j/graphql/issues/1756, nested bug", () => {
```

Comment:
:thinking: It seems that this shares the type definitions, maybe it makes sense to have this as a new test case within the same `describe` instead of a separate one?